src/js/validation.js

```diff
@@ -187,10 +187,11 @@ export function isValidForm(form, pageListByChapters) {
     .filter(pageKey => isActivePage(_.find({ pageKey }, pageConfigs), form.data));
 
   const v = new Validator();
+  const fullFormData = form.data;
```

Comment:
(Me, on my regular account since I don't have my work computer with me.)

For `showPagePerItem` array pages, [this line](https://github.com/usds/us-forms-system/blob/6b291209e34a5eb4ffccd69a59b741cf538d44e7/src/js/validation.js#L199) alters `formData` so that it only represents the data on the page with the current index.  So by the time we we [call the validator](https://github.com/usds/us-forms-system/blob/6b291209e34a5eb4ffccd69a59b741cf538d44e7/src/js/validation.js#L213) a few lines down the validator is only seeing that part of the whole form.  By saving a reference with `fullFormData` we can pass it along as an extra final param, and it will will be the same as `formData` for all but array pages.

I'm not sure offhand of a better design for this but would definitely like to see one because, yeah, it's a mess.

Reply:
It could be done that way but @jbalboni  thought it might break quite a few forms since they currently only expect the current page for that case. Also I don't think the index is currently passed is it? 

Since this adds the full data as a new final arg it's backwards compatible so it can go out in a 1.x release. If a different design is placed in a major release I suspect it won't be usable by vets.gov for quite a while. In any case I think we can almost guarantee this will change to some other approach in 2.x or later since it's pretty awkward as-is.

Reply:
@annekainicUSDS Yeah, this validation code shouldn't really be getting the data filtered by index. I think that was a mistake when we wrote this initially (or there's some obscure reason that I've forgotten). But like Dave says, it would be a breaking change.

Reply:
Gotcha, okay now I understand, sorry I was slow to catch up! I'd be curious to know, if we were to make the breaking change, what work would need to happen to fix that on the Vets.gov side?

Reply:
We'd have to look through the forms that use pagePerItem and find all the custom validations in those pages. That's I think three forms at the moment.

Reply:
Thanks @jbalboni! I think we can merge this PR for now and then @dmethvin how would you feel about revisiting this code later and making a more permanent fix that might be a breaking change, but I think would make things clearer?

Reply:
Yeah I think that would be the way to go.